File: EDUARDO/Integrador.m
%
classdef Integrador < handle
   properties
        Vx
        Vy
        Vz
        Sx 
        Sy
        Sz
   end
   methods
        function obj =Integrador(initval)
            obj.Vx=initval;
            obj.Vy=initval;
            obj.Vz=initval;

            obj.Sx=initval;
            obj.Sy=initval;
            obj.Sz=initval;
        end
        function integra_rt(obj,dt,a_n2,Db)

            obj.Vx = obj.Vx+a_n2(1)*dt;
            obj.Vy = obj.Vy+a_n2(2)*dt;
            obj.Vz = obj.Vz+a_n2(3)*dt;			


            obj.decide(Db);
					
            obj.Sx = obj.Sx + obj.Vx*dt;
            obj.Sy = obj.Sy + obj.Vy*dt;
            obj.Sz = obj.Sz + obj.Vz*dt;
        end

        function decide(obj,Db)

            if (Db == 1)
        
                obj.Vx=0;
                obj.Vy=0;
                obj.Vz=0;
        
            end
    

        end

   end
end

File: EDUARDO/JAVA/AppCompleto_v12/app/src/main/java/fun/eduardo/funcoes/Integrador.java
package fun.eduardo.funcoes;

import net.trucomanx.pdsplibj.pdsra.*;

public class Integrador {

	public double Vx,Vy,Vz,Sx,Sy,Sz;


	public Integrador(int initval){

		this.Vx = initval;
		this.Vy = initval;
		this.Vz = initval;

		this.Sx = initval;
		this.Sy = initval;
		this.Sz = initval;

	}	


	//a_n2: input signal  [1x3]
    //dt: input time interval [1x1]
    //Db: input saída Detector de pausa 0 [movimento] ou 1 [pausa]
    //return :::
    //Sx: Deslocamento no eixo x [1x1]
    //Sy: Deslocamento no eixo y [1x1]
    //Sz: Deslocamento no eixo z [1x1]
    //Vx: Deslocamento no eixo x [1x1]
    //Vy: Deslocamento no eixo y [1x1]
    //Vz: Deslocamento no eixo z [1x1]


   	public double integra_rt(double dt, PdsMatrix a_n2, double Db) {   

		this.Vx = this.Vx+a_n2.GetValue(0,0)*dt;
        this.Vy = this.Vy+a_n2.GetValue(0,1)*dt;
        this.Vz = this.Vz+a_n2.GetValue(0,2)*dt;

		if (Db == 1)
        {
        	this.Vx=0;
        	this.Vy=0;
            this.Vz=0;
        
        }		

		this.Sx = this.Sx + this.Vx*dt;
        this.Sy = this.Sy + this.Vy*dt;
        this.Sz = this.Sz + this.Vz*dt;
            
        return (Sx);
		
	}

	public double get_Vx(){
		return(Vx);
	}

	public double get_Vy(){
		return(Vy);
	}

	public double get_Vz(){
		return(Vz);
	}


	public double get_Sx(){
		return(Sx);
	}

	public double get_Sy(){
		return(Sy);
	}

	public double get_Sz(){
		return(Sz);
	}




	
}





